sympy/physics/control/lti.py

```diff
@@ -0,0 +1,655 @@
+from sympy import Basic, Mul, degree, Symbol, expand, cancel, Expr, exp
+from sympy.core.evalf import EvalfMixin
+from sympy.core.numbers import Integer
+from sympy.core.sympify import sympify, _sympify
+
+__all__ = ['TransferFunction', 'Series', 'Parallel', 'Feedback']
+
+
+class TransferFunction(Basic, EvalfMixin):
+    """TransferFunction(num, den, var)
+
+    A class for representing continuous transfer functions. This class is used to represent
+    LTI (Linear, time-invariant) systems in transfer function form. The arguments
+    are ``num``, ``den``, and ``var``, where ``num`` and ``den`` are numerator and
+    denominator polynomials of the ``TransferFunction`` respectively, and the third argument is
+    a variable used to anchor these polynomials of the transfer function. ``num`` and ``den`` can
+    be either sympy expressions (with no time delay terms) or numbers, whereas ``var``
+    has to be a Symbol.
```

Comment:
I don't see how this is true. How can we allow any sympy expression?

Reply:
@moorepants, a few days before you asked me to add unit tests for transfer functions whose numerator or denominator can have a float exponent. I don't think a numerator like `a0*s**0.5 + a2*s**0.6 - a1` can be called a "polynomial" altho it comes into the category of an `Expr`, that's why I have written "polynomial/expression" everywhere.

Reply:
It should have an integer exponent to be called as a "polynomial"!

Reply:
I can't remember what I wrote a few days ago, but I assume I must have meant to write unit tests for error checking to prevent float exponents. There are two places to think about exponents 1) the numerator and denominator expressions and 2) pow(TransferFunction, n). I don't think floats for either of these makes much sense.

This class should only support valid expressions resulting from Laplace Transforms. An for now I think it can only support polynomials of s that have integer values. I don't think we can support other possible expressions (yet) so we should raise an error if a user creates a TransferFunction with anything other than a polynomial, no?

Reply:
Although it is not common, there are situations when transfer functions can be non-rational.  These often arise when dealing with PDEs, for instance.  Some examples [here](http://www.math.uwaterloo.ca/~kmorris/Preprints/Curtain_Morris_final.pdf).

Reply:
Thanks @murrayrm. We can support them but it may be easier to support a strict subset now and add more expression types in the future as we understand them more and the use cases. If we support rational and irrational polynomials now, it probably doesn't make much difference on the development side of things, so we could leave it in.

Reply:
I agree with @murrayrm. We should support non-rational transfer functions now only. @moorepants, I think there is only one way to support only "polynomials", and that is by introducing `Poly` objects internally. The whole class structure would have to be changed I think!

Reply:
However, I think that it is too trivial and not practically useful to restrict for integer coefficients. It should be able to support any symbolic real numbers.

Reply:
> @moorepants, I think there is only one way to support only "polynomials", and that is by introducing Poly objects internally. The whole class structure would have to be changed I think!

That may be the case. That's why I asked at the very beginning to investigate whether we should be using Poly internally.

I'm supporting only expressions that are composed of powers of s. It'll be garbage in, garbage out if users supply things that aren't supported. Which is a fine design decision as long as the docstring warns about this.

Reply:
However, I'd note that it is *not* always an option to use Poly to restrict for integer exponents.

![image](https://user-images.githubusercontent.com/34944973/86524029-14c1fb00-beb0-11ea-9fb9-551e4f2d537a.png)


For example, we can use other mathematical objects like multiplication or `Product` to describe a transfer function given by roots, while Poly is actually not an efficient option for this because it automatically expands everything.

Reply:
@murrayrm 
Is there any transfer function in the form of polynomials of modular coefficients or finite fields?
If there exists rigid theory about that, it gives a suitable reason for using Poly.